h2o-extensions/xgboost/src/test/java/hex/tree/xgboost/XGBoostVarImpComparisonTest.java

```diff
@@ -0,0 +1,112 @@
+            {"gbtree", "gamma", "AGE"},
+            {"gbtree", "poisson", "AGE"},
+            {"gbtree", "tweedie", "AGE"},
+            {"dart", "AUTO", "AGE"},
+            {"dart", "bernoulli", "CAPSULE"},
+            {"dart", "multinomial", "CAPSULE"},
+            {"dart", "gaussian", "AGE"},
+            {"dart", "gamma", "AGE"},
+            {"dart", "poisson", "AGE"},
+            {"dart", "tweedie", "AGE"},
+            {"gblinear", "AUTO", "AGE"},
+            {"gblinear", "bernoulli", "CAPSULE"},
+            {"gblinear", "multinomial", "CAPSULE"},
+            {"gblinear", "gaussian", "AGE"},
+            {"gblinear", "gamma", "AGE"},
+            {"gblinear", "poisson", "AGE"},
+            {"gblinear", "tweedie", "AGE"}
+        });
+    }
+
+    @Parameterized.Parameter
+    public String booster;
+
+    @Parameterized.Parameter(1)
+    public String distribution;
+
+    @Parameterized.Parameter(2)
+    public String response;
+
+    @Test
+    public void testImportancesAreSame() {
+        Scope.enter();
+        try {
+            Frame tfr = Scope.track(parse_test_file("./smalldata/prostate/prostate.csv"));
+            // define special columns
+            Scope.track(tfr.replace(1, tfr.vecs()[1].toCategoricalVec()));   // Convert CAPSULE to categorical
+            Scope.track(tfr.replace(3, tfr.vecs()[3].toCategoricalVec()));   // Convert RACE to categorical
+            DKV.put(tfr);
+
+            // split into train/test
+            SplitFrame sf = new SplitFrame(tfr, new double[]{0.7, 0.3}, null);
+            sf.exec().get();
+            Key[] splits = sf._destination_frames;
+            Frame trainFrame = Scope.track((Frame) splits[0].get());
+            Frame testFrame = Scope.track((Frame) splits[1].get());
+
+            XGBoostModel.XGBoostParameters parms = new XGBoostModel.XGBoostParameters();
+            parms._booster = XGBoostModel.XGBoostParameters.Booster.valueOf(booster);
+            parms._distribution = DistributionFamily.valueOf(distribution);
+            parms._ntrees = 10;
+            parms._max_depth = 5;
+            parms._train = trainFrame._key;
+            parms._valid = testFrame._key;
+            parms._response_column = response;
+
+            System.setProperty("sys.ai.h2o.xgboost.predict.native.enable", "true");
+            XGBoostModel nativeModel = new XGBoost(parms).trainModel().get();
+            Scope.track_generic(nativeModel);
+
+            System.setProperty("sys.ai.h2o.xgboost.predict.native.enable", "false");
+            XGBoostModel javaModel = new XGBoost(parms).trainModel().get();
+            Scope.track_generic(javaModel);
+
+            if (nativeModel._output._varimp != javaModel._output._varimp) {
+                assertArrayEquals(nativeModel._output._varimp._names, javaModel._output._varimp._names);
+                assertArrayEquals(nativeModel._output._varimp._varimp, javaModel._output._varimp._varimp, 1e-2f);
```

Comment:
Why do we need a large tolerance? What differs in their math compared to ours? float vs doubles?

Reply:
float only has 6-7 digits for precision for numbers of 1234.56 there can be a difference on the 0.01 scale probably because of order of additions